### PR TITLE
Add CI pipeline for building and pushing multi-arch Docker image

### LINKED ISSUE #1428: Add native support for Apple M1 in Docker Images
**Describe the feature request**
It would be very helpful if Mockserver could push Docker images that support the arm64 architecture.

**What you are trying to do**
I am not familiar with the release process. I have not found any automatism for it in the project. Otherwise I would have already provided a pull request. But please see the next section, there is the solution described.

**The solution you'd like**
It would be nice if the current docker build command could swap with the following one:

`docker buildx build --platform=linux/arm64,linux/amd64 --no-cache --tag mockserver/mockserver:mockserver-5.13.2 --push ./docker && docker buildx build --platform=linux/arm64,linux/amd64 --no-cache --tag mockserver/mockserver:5.13.2 --push ./docker`

I have already executed and tested this on several projects. Some references: https://hub.docker.com/u/ezienecker

**Describe alternatives you've considered**
There is no alternative.

### PR DESCRIPTION
**Pull request by dennisameling**
This PR adds a CI pipeline for automatically building and pushing a Docker image to Docker Hub on every GitHub tag that is created.

See it in action here: https://github.com/dennisameling/mockserver/runs/7864265832?check_suite_focus=true (published to https://hub.docker.com/r/dennisameling/mockserver/tags)

Closes #1428 and https://trello.com/c/fUHwzasD/153-add-arm64-docker-image-for-hosting-on-raspberrypi-devices

### CODE REVIEW

File: .github/workflows/build-docker-image.yml [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+        run: echo "::set-output name=version::${GITHUB_REF/refs\/tags\//}"
+      - name: Set up QEMU
+        uses: docker/setup-qemu-action@v2
+      - name: Set up Docker Buildx
+        uses: docker/setup-buildx-action@v2
+      - name: Login to DockerHub
+        uses: docker/login-action@v2
+        with:
+          username: ${{ secrets.DOCKERHUB_USERNAME }}
+          password: ${{ secrets.DOCKERHUB_TOKEN }}
```
Comment (dennisameling) [inline .github/workflows/build-docker-image.yml:23]: You'd need to set both of these secrets in the repo's GitHub Actions settings.

File: .github/workflows/build-docker-image.yml [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+      - name: Build and push
+        uses: docker/build-push-action@v3
+        with:
+          context: .
+          file: docker/Dockerfile
+          platforms: linux/amd64,linux/arm64
+          push: true
+          tags: mockserver/mockserver:${{ steps.get_tag_version.outputs.version }}
```
Comment (dennisameling) [inline .github/workflows/build-docker-image.yml:31]: This would e.g. create `mockserver/mockserver:mockserver-5.13.2` in Docker Hub

**Comment by jamesdbloom** [comment]
This looks good I'll try and take a look later today

**Comment by jamesdbloom** [comment]
Sadly due to manual process in nexus to approve a release (which I don't believe can be automated by design from nexus) the build always picks the previous version.  But a manual re-run of the job fixes it luckily.  This is pretty lame but I guess for now it'll have to do.

**Pull request merged**